mne/preprocessing/ica.py

```diff
@@ -2002,11 +2008,32 @@ def find_bads_muscle(
             psds = psds.mean(axis=0)
         slopes = np.polyfit(np.log10(freqs), np.log10(psds).T, 1)[0]
 
+        # typical muscle slope is ~0.15, non-muscle components negative
+        # so logistic with shift -0.5 and slope 0.25 so -0.5 -> 0.5 and 0->1
+        slope_score = expit((slopes + 0.5) / 0.25)
+
+        # Need sensor positions for the criteria below, so return with only one score
+        # if no positions available
+        picks = _picks_to_idx(
+            inst.info, self.ch_names, "all", exclude=(), allow_empty=False
+        )
+        if not _check_ch_locs(inst.info, picks=picks):
+            logger.warning(
+                "No sensor positions found. Scores for bad muscle components are only "
+                "based on the 'slope' criterion."
+            )
```

Comment:
For this you should use `warn` from `mne.utils`, it does some nice stuff based on the logging level etc.